ambry-router/src/main/java/com.github.ambry.router/ValidatedByteRange.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.router;
+
+/**
+ * a byte range with defined start and end offsets that are verified to be within the supplied total blob size.
+ */
+class ValidatedByteRange {
```

Comment:
I don't feel like we should mutate the original `ByteRange`, in case the caller that provided it wants to do something with the original offsets.  However, I could move the validation/offset setting logic into `GetBlobOperation`, and store the fully defined offsets in a `ByteRange` member variable.